crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs

```diff
@@ -24,8 +24,14 @@ impl BuiltInResult {
         // TODO: Choose the kind of Result representation based on whether or not the ok and error
         //  types are primitives.
         //  See `swift-bridge/src/std_bridge/result`
-        let result_kind = quote! {
-            ResultPtrAndPtr
+        let result_kind = if self.ok_ty.is_null() {
```

Comment:
Can use `self.ok_ty.has_exactly_one_encoding()` so that this works for types like `struct UnitStruct;`

I actually plan to rename `.has_exactly_one_encoding()` -> `.can_be_encoded_with_zero_bytes()`. 
Feel free to rename it while you're here if you like.